pandas/io/common.py

```diff
@@ -92,6 +91,34 @@ def _is_url(url):
         return False
 
 
+def _urlopen(url, session=None):
+    compression = None
+    content_encoding = None
+    try:
+        import requests
```

Comment:
There is no need to check if `requests` is installed when using a `session` object b/c `requests` is needed to create the object, this is slitly simpler than the approach in #17087 and introduces less code to maintain.